src/Psl/Tree/from_array.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Tree;
+
+use Psl\Vec;
+
+/**
+ * Creates a tree from a nested array structure.
+ *
+ * Example:
+ *
+ *      Tree\from_array([
+ *          'value' => 'root',
+ *          'children' => [
+ *              ['value' => 'child1', 'children' => []],
+ *              ['value' => 'child2', 'children' => []],
+ *          ],
+ *      ])
+ *
+ * @template T
+ *
+ * @param array{
+ *   value: T,
+ *   children?: list<array{
+ *     value: T,
+ *     children?: list<array{
+ *       value: T,
+ *       children?: list<array{
+ *         value: T,
+ *         children?: list<array{
+ *           value: T,
+ *           children?: list<array>
+ *         }>
+ *       }>
+ *     }>
+ *   }>
+ * } $array
+ *
+ * @return TreeNode<T>
+ *
+ * @pure
+ */
```

Comment:
What would be a nice addition is something like a `from_(vec|list)` with a grouping function.
For example to create a tree based on db records that are linked by using a parent_id. (if the parent_id = null, you have to root and can start filling up from there)

wdyt?